app/src/androidTest/java/com/google/jetpackcamera/UiTestUtil.kt

```diff
@@ -37,6 +38,15 @@ inline fun <reified T : Activity> runScenarioTest(
     }
 }
 
+inline fun <reified T : Activity> runScenarioTestForResult(
+    intent: Intent,
+    crossinline block: ActivityScenario<T>.() -> Unit
+): ActivityScenario<T> {
```

Comment:
Could you just have this return `ActivityResult` instead of `ActivityScenario`? Just put in a `return scenario.result` after `scenario.apply(block)`.